insights/core/spec_factory.py

```diff
@@ -142,18 +143,20 @@ def validate_lines(self, results):
         return True
 
     def load(self):
-        # read with universal newlines so all line terminators are converted to
-        # "\n"
-        with open(self.path, 'U') as f:
-            results = []
-            if self.ds:
-                # This should shell out to a grep pipeline
-                results = [l.rstrip("\n") for l in apply_filters(self.ds, f)]
+
+        filters = False
+        if self.ds:
+            filters = "|".join(get_filters(self.ds))
+        if filters:
+            rc, out = subproc.call("grep -E '{0}' {1} ".format(filters, self.path), shell=False, keep_rc=True)
+            if rc == 0 and out != '':
+                results = [l.rstrip() for l in out.splitlines()]
```

Comment:
I think we just changed the behavior of datasources to *not* `rstrip`.

Reply:
Hey @jhjaggars ,
I noticed that in master there is still one instance of rstrip in the load() function in TextFileProvider class in the master branch.
Do I still need to make the changes to remove the rstrip()?

https://github.com/RedHatInsights/insights-core/blob/f99b9fccb9fece4fa8083505a29e77bf432a31f6/insights/core/spec_factory.py#L149-L151

Reply:
Calling @csams.

Reply:
I think we should remove the strip and test the change.  It seems weird to keep this one while all the others are taken out.  Testing the change should show us if it has adverse affects on rule hits.

Reply:
I will remove the strip from the new code which should take care of all of the rstrips() in the spec_fatctory.